shared/domains/chelonia/SPMessage.js

```diff
@@ -458,6 +461,8 @@ export class SPMessage {
 
   hash (): string { return this._mapping.key }
 
+  previousKeyOp (): string { return this._head.previousKeyOp }
```

Comment:
o4-mini-high points out:

> In your new `previousKeyOp(): string { … }` you probably want to return `?string` since it may be `null` on very old messages.  

I think this is correct as `OP_CONTRACT` would have a `null` previousOp